enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/grpc/ExtAuthService.java

```diff
@@ -137,11 +136,32 @@ private CheckResponse buildResponse(CheckRequest request, ResponseObject respons
                         structBuilder.putFields(key, Value.newBuilder().setStringValue(value).build()));
             }
 
-            return CheckResponse.newBuilder()
+            CheckResponse.Builder checkResponseBuilder = CheckResponse.newBuilder()
                     .setStatus(Status.newBuilder().setCode(getCode(responseObject.getStatusCode())))
-                    .setDeniedResponse(responseBuilder.setBody(responseJson.toString()).setStatus(status).build())
-                    .setDynamicMetadata(structBuilder.build())
-                    .build();
+                    .setDynamicMetadata(structBuilder.build());
+
+            if (!isMockedApi) {
```

Comment:
isn't it better to handle mockedapis check inside restapi.java's process() method